From 19e1c61d09bb2c6637e79e88057da43ef8b37a18 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Tue, 16 Jul 2024 18:00:32 -0700
Subject: [PATCH] mm/damon/dbgfs-test: skip dbgfs_set_targets() test if PADDR
 is not registered

The test depends on registration of DAMON_OPS_PADDR, but that's not what
DAMON debugfs interface can control.  Just skip the test if it is not
registered.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/dbgfs-test.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mm/damon/dbgfs-test.h b/mm/damon/dbgfs-test.h
index 2d85217f5ba4..9bd5dca5d4ad 100644
--- a/mm/damon/dbgfs-test.h
+++ b/mm/damon/dbgfs-test.h
@@ -73,6 +73,11 @@ static void damon_dbgfs_test_set_targets(struct kunit *test)
 	struct damon_ctx *ctx = dbgfs_new_ctx();
 	char buf[64];
 
+	if (!damon_is_registered_ops(DAMON_OPS_PADDR)) {
+		dbgfs_destroy_ctx(ctx);
+		kunit_skip(test, "PADDR not registered");
+	}
+
 	/* Make DAMON consider target has no pid */
 	damon_select_ops(ctx, DAMON_OPS_PADDR);
 
-- 
2.39.2

